### DISCUSSION TITLE
Assign RFID Code to an employee

### DISCUSSION THREAD

**Discussion by Den4ik1992** [Budibase/budibase#10348] [Help] [answered]
Hello everyone, in advance I am not an extremely experienced Budibase user.

I have an RFID scanner at the entrance of the production facility where employees can clock in and out at the beginning and end of their shifts for timekeeping purposes.

The ID from the chip is directly written into a table with the date and time using the Budibase webhook trigger (Python Script - POST Request). However, I also want to automatically assign a name to the stamp number in the table. I have a database where the name is listed with the corresponding stamp number.

Now, I want the name of the person who stamped their card to be automatically written into the table with the webhook.

![2023-04-19 12_03_31-Budibase und 5 weitere Seiten - Geschäftlich – Microsoft​ Edge](https://user-images.githubusercontent.com/82113641/233041672-58d81d14-ea9e-41d3-b5dc-f045414d7633.png)

![2023-04-19 12_03_07-Budibase und 5 weitere Seiten - Geschäftlich – Microsoft​ Edge](https://user-images.githubusercontent.com/82113641/233041679-778357b4-c452-4bbc-87b9-cdc28367c8ce.png)

I think it is certainly simple to solve but I do not get on, thanks for the help in advance !

**Comment by ConorWebb96** [comment, marked as answer]
Hey @Den4ik1992, 

I'm assuming you are using the Budibase data source. As I can't see the automation you are running, I'll describe what needs to be done as best I can.

First in your employees table can you make sure the Employee column has the toggle switched on for 
_Use as table display column_. This can be done by hovering over the column name within the table the pencil icon. (Screenshot below.)
<img width="1023" alt="Screenshot 2023-04-19 at 11 26 22" src="https://user-images.githubusercontent.com/126772285/233047143-2cb1d1d1-6311-4a1a-8129-e2e3f62ed8cf.png">

Click this and it should open a modal window, you can find the toggle _Use as table display column_ here click it and then save.
<img width="413" alt="Screenshot 2023-04-19 at 11 27 41" src="https://user-images.githubusercontent.com/126772285/233047453-d92a6cc6-22f4-4743-9eeb-e1080b7a3ed8.png">

From here, I'd recomend deleting the employee column in your main table and re-adding it as a relationship field. I'd suggest a one to many relationship, where one employee can have many _Time Entries_ (Not sure on what your main table is called). 

After the above, lets head back to your automation with the web-hook.

1. Create a new step with the query row at the start of the automation.
<img width="483" alt="Screenshot 2023-04-19 at 12 51 11" src="https://user-images.githubusercontent.com/126772285/233066420-3c781356-e14f-4d24-a8fd-5c5d87b63f32.png">
2. Customise the query to suit your needs, normally i'd, limit the query row to 1, return nothing if its empty.
<img width="487" alt="Screenshot 2023-04-19 at 13 03 47" src="https://user-images.githubusercontent.com/126772285/233069726-48052b90-f331-4bef-8748-52dd4f478e80.png">

4. Add a filter to get the specific Employee based on the id received from your web-hook. 
<img width="1662" alt="Screenshot 2023-04-19 at 12 34 52" src="https://user-images.githubusercontent.com/126772285/233066654-1d9f396e-1f47-4898-b3a7-1a3d3f005f11.png">

5. (Optional) Can add in a conditional statement to stop the automation if nothing is returned.
6. Modify your existing row creation action for the Time entries table, ensure bindings is selected.
<img width="499" alt="Screenshot 2023-04-19 at 12 57 35" src="https://user-images.githubusercontent.com/126772285/233067803-32ce9903-3318-4874-b9eb-bfa2b096146b.png">

7. From here attach the id of the employee based on the queried row step at the beginning.

Test and save and this should do what you want!

**Optionally** you don't have to create relationships between the tables. You could do a similar process without relationships and take the string name of the employees table and add this into the text field you currently have.

Hopefully this makes sense and helps you!

**Reply by Den4ik1992** [reply]
Thank you for the extremely fast and detailed answer. It’s simply incredible what a great and helpful community Budibase has.

I took the simple method for me without relationship and it works just perfect. With relationship I find it still a bit complicated because I have never worked with databases. Must still practice something :)

If I get stuck somehow I might need some help. But I think I can manage the rest already ;)

Thanks !

**Reply by ConorWebb96** [reply]
Perfect glad to hear its now working for you!
